src/main/java/fr/sncf/osrd/infra/TVDSectionPath.java

```diff
@@ -40,6 +41,10 @@ public EdgeDirection getStartTrackDirection() {
         return trackSections[0].direction;
     }
 
+    public boolean contains(TrackSectionLocation location) {
+        return Arrays.stream(trackSections).anyMatch(range -> range.containsLocation(location));
```

Comment:
I'm thinking there might be a corner case where the range isn't contained by any individual range, but is contained by the union of all ranges